bin/ivar_variants_to_vcf.py

```diff
@@ -146,71 +146,92 @@ def make_dir(path):
-        '##FORMAT=<ID=REF_RV,Number=1,Type=Integer,Description="Depth of reference base on reverse reads">\n'
-        '##FORMAT=<ID=REF_QUAL,Number=1,Type=Integer,Description="Mean quality of reference base">\n'
-        '##FORMAT=<ID=ALT_DP,Number=1,Type=Integer,Description="Depth of alternate base">\n'
-        '##FORMAT=<ID=ALT_RV,Number=1,Type=Integer,Description="Depth of alternate base on reverse reads">\n'
-        '##FORMAT=<ID=ALT_QUAL,Number=1,Type=String,Description="Mean quality of alternate base">\n'
-        '##FORMAT=<ID=ALT_FREQ,Number=1,Type=Float,Description="Frequency of alternate base">\n'
-    )
-    header += (
-        "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t" + filename + "\n"
-    )
-
-    varList = []
-    varCountDict = {"SNP": 0, "INS": 0, "DEL": 0}
-    dict_lines = {'CHROM':[],'POS':[],'ID':[],'REF':[],'ALT':[],'REF_DP':[],'REF_RV':[],'ALT_DP':[],'ALT_RV':[],'QUAL':[],'REF_CODON':[],'ALT_CODON':[],'FILTER': [],'INFO':[],'FORMAT':[],'SAMPLE':[]}
-    writeLine=False
-    OutDir = os.path.dirname(FileOut)
-    make_dir(OutDir)
-    fout = open(FileOut, "w")
-    fout.write(header)
-
-    with open(FileIn) as f:
-        for line in f:
+    ## Create output directory
+    filename = os.path.splitext(file_in)[0]
+    out_dir = os.path.dirname(file_out)
+    make_dir(out_dir)
+
+    ## Define VCF header
+    header_source = [
+        "##fileformat=VCFv4.2",
+        "##source=iVar"
+    ]
+    header_info = [
+        '##INFO=<ID=DP,Number=1,Type=Integer,Description="Total Depth">'
+    ]
+    header_filter = [
+        '##FILTER=<ID=PASS,Description="Result of p-value <= 0.05">',
+        '##FILTER=<ID=FAIL,Description="Result of p-value > 0.05">'
+    ]
+    header_format = [
+        '##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">',
+        '##FORMAT=<ID=REF_DP,Number=1,Type=Integer,Description="Depth of reference base">',
+        '##FORMAT=<ID=REF_RV,Number=1,Type=Integer,Description="Depth of reference base on reverse reads">',
+        '##FORMAT=<ID=REF_QUAL,Number=1,Type=Integer,Description="Mean quality of reference base">',
+        '##FORMAT=<ID=ALT_DP,Number=1,Type=Integer,Description="Depth of alternate base">',
+        '##FORMAT=<ID=ALT_RV,Number=1,Type=Integer,Description="Depth of alternate base on reverse reads">',
+        '##FORMAT=<ID=ALT_QUAL,Number=1,Type=String,Description="Mean quality of alternate base">',
+        '##FORMAT=<ID=ALT_FREQ,Number=1,Type=Float,Description="Frequency of alternate base">',
+    ]
+    header_cols = [
+         f"#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t{filename}"
+    ]
+    if not ignore_strand_bias:
+        header_info += [
+            '##INFO=<ID=SB,Number=1,Type=Float,Description="Strand-bias fisher-test p-value">'
+        ]
+        header_filter += [
+            '##FILTER=<ID=SB,Description="Strand-bias fisher-test p-value < 0.05">'
```

Comment:
From Slack:

Me: 
Should this be `p-value > 0.05`?

Sara: 
Mm, let's see if I'm wrong the fisher_test evaluates if the forward and reverse distribution is different, which is the strand-bias, so if p-value < 0.05 is when the SB filter is set.

Me:
Ok. Just double-checking. Normally, you would set these sorts of things if the p-value is NOT significant.

Reply:
we've inspired ourselves in this [gatk article](https://gatk.broadinstitute.org/hc/en-us/articles/360035532152-Fisher-s-Exact-Test), according to it:

> Our null hypothesis is that there is no difference in the number of reads that support the reference allele and alternate allele on each strand (there is no strand bias). We will calculate a [p-value](http://mathworld.wolfram.com/P-Value.html) that tells us the probability of observing our data if our null hypothesis is true (or, that there is no strand bias). The lower the p-value, the less likely we are to believe that there is no strand bias.

So we set as there is strand-bias when p-value < 0.05.

Reply:
Ok 👍🏽